tt_eager/tt_dnn/op_library/bmm/bmm_op.cpp

```diff
@@ -692,14 +692,17 @@ Tensor falcon_dense_4h_to_h_matmul(const Tensor &input_tensor_a, const Tensor &i
 Tensor falcon_dense_h_to_4h_matmul(const Tensor &input_tensor_a, const Tensor &input_tensor_b, std::optional<const Tensor> bias, std::optional<UnaryWithParam> fused_activation, const MemoryConfig& mem_config, std::optional<const DataType> output_dtype) {
     auto seq_len = input_tensor_a.shape()[2];
     if (seq_len > 1024) {
-        TT_FATAL(not fused_activation.has_value());
         // TODO: Check support for seq_len == 128, 256, 512, ..., 2048
         TT_FATAL(seq_len % TILE_HEIGHT == 0, "Falcon mm's seq_len must be a multiple of 32!");
         TT_FATAL(seq_len >=  128, "Falcon mm's seq_len must be greater than 128!");
         TT_FATAL((input_tensor_a.shape() == Shape({1, 1, seq_len, 4544})), "Unsupported input shape");
         TT_FATAL((input_tensor_b.shape() == Shape({1, 1, 4544, 18176})), "Unsupported input shape");
-        TT_FATAL(!fused_activation.has_value());
-        return operation::run_with_autoformat(Matmul{.bcast_batch=true, .output_mem_config=mem_config, .output_dtype=output_dtype.value_or(input_tensor_a.dtype())}, {input_tensor_a, input_tensor_b}).at(0);
+        Tensor mm = operation::run_with_autoformat(Matmul{.bcast_batch=true, .output_mem_config=mem_config, .output_dtype=output_dtype.value_or(input_tensor_a.dtype())}, {input_tensor_a, input_tensor_b}, {std::nullopt}).at(0);
```

Comment:
Is this affecting prefill and decode?
If so, can we still use fused matmul + gelu for decode?

Reply:
Yes, it will work as expected. Ideally we want to fuse gelu inside kernel, which currently is not supported.

Reply:
GELU is fused with MM in BERT, is the config is different here to prevent fusing? FYI @TT-BrianLiu

Reply:
Looks like ttnn.linear (which takes gelu as activation too) is using this here under to hood:

```
program_config = ttnn.ttl.operations.primary.MatmulMultiCoreReuseMultiCastProgramConfig(
                compute_with_storage_grid_size=(core_grid.x, core_grid.y),
                per_core_M=y_tiles_per_core,
                per_core_N=x_tiles_per_core,
                in0_block_w=k_block_size,  # k
                out_subblock_h=m_subblock_size,  # m
                out_subblock_w=n_subblock_size,  # n
                transpose_mcast=False,
                fused_activation=fused_activation,
            )
            
output_tensor = ttnn.ttl.operations.primary.matmul(
                input_tensor_a,
                input_tensor_b,
                bias=bias,
                program_config=program_config,
                output_mem_config=memory_config,
                output_dtype=dtype,
            )
```

So we could just use 
`tt_lib.operations.primary.matmul` with the correct `program_config` incl. the `fused_activation`, right?

Reply:
run_with_autoformat, should choose specific matmul, but it turns out that for our dimension, it will take basic implementation which does not have fusing implemented inside kernel.  I am playing with dimensions, to enable MatmulMultiCoreReuse, but currently I am getting errors where L1 is out of space.